---
# Pin definitions
#
# output_pins: are read from a topic (default) or service

# HAL pins are named hal_io.<pin_name>

# valid keys for all pins:
#   hal_type:  BIT, FLOAT, S32, U32

publish_pins:
  # publish_pins values from HAL pin published to ROS topic
  # valid keys:
  #   hal_dir:  IN (default), IO
  #   pub_topic:  (any valid ROS topic name, default /hal_io/<pin_name>)
  din01:
    hal_type: BIT
  din02:
    hal_type: BIT
  din03:
    hal_type: BIT
  din04:
    hal_type: BIT
  din05:
    hal_type: BIT
  din06:
    hal_type: BIT
  din07:
    hal_type: BIT
  din08:
    hal_type: BIT
  din09:
    hal_type: BIT
  din10:
    hal_type: BIT
  din11:
    hal_type: BIT
  din12:
    hal_type: BIT
  din13:
    hal_type: BIT
  din14:
    hal_type: BIT
  din15:
    hal_type: BIT
  din16:
    hal_type: BIT
subscribe_pins:
  # subscribe_pins values from ROS topic copied to HAL output pin
  # valid keys:
  #   hal_dir:  OUT (default), IO
  #   sub_topic:  (any valid ROS topic name, default /hal_io/<pin_name>)
  #   pub_topic:  (any valid ROS topic name, default /hal_io/<pin_name>)
  dout01:
    hal_type: BIT
  dout02:
    hal_type: BIT
  dout03:
    hal_type: BIT
  dout04:
    hal_type: BIT
  dout05:
    hal_type: BIT
  dout06:
    hal_type: BIT
  dout07:
    hal_type: BIT
  dout08:
    hal_type: BIT
  dout09:
    hal_type: BIT
  dout10:
    hal_type: BIT
  dout11:
    hal_type: BIT
  dout12:
    hal_type: BIT
  dout13:
    hal_type: BIT
  dout14:
    hal_type: BIT
  dout15:
    hal_type: BIT
  dout16:
    hal_type: BIT
  control_mode:
    hal_type: S32
    initial_value: 8   # CSP

service_pins: {}
  # service_pins values from ROS service copied to HAL output pin
  #   and published topic
  # valid keys:
  #   hal_dir:  OUT (default), IO
  #   service_name:  (any valid ROS service name, default /hal_io/<pin_name>)
  #   pub_topic:  (any valid ROS topic name, default /hal_io/<pin_name>)
